File: app/src/main/java/com/perc/pavel/sportgeolocationgame/TcpListener.java
package com.perc.pavel.sportgeolocationgame;


import org.json.JSONException;
import org.json.JSONObject;

/**
 * Интерфейс для получения ответов от tcp сервера
 */
interface TcpListener {
    /**
     * Вызывается при ответе сервера.
     * @param message Ответ сервера.
     */
	void onTCPMessageReceived(JSONObject message);

//    /**
//     * Вызывается при подключении или отключении от сервера.
//     * @param isConnectedNow True - сервер подключён, False - сервер отключён.
//     */
//	void onTCPConnectionStatusChanged(boolean isConnectedNow);
//    
    
    void onConnected();
    void onDisconnected();
}

File: app/src/main/java/com/perc/pavel/sportgeolocationgame/serverworking/TcpListener.kt
package com.perc.pavel.sportgeolocationgame.serverworking

import org.json.JSONObject

/** Interface for receiving callbacks from tcp server*/
interface TcpConnectionListener {
    fun onConnected()
    fun onConnectionError(error: String, title: String = "")
}

interface TcpMessageListener {
    /**
     * Is invoked when the server responses
     * @param message Server response
     */
    fun onTCPMessageReceived(message: JSONObject)
}
